x-pack/plugins/apm/server/routes/transactions/route.ts

```diff
@@ -364,10 +365,64 @@ const transactionChartsErrorRateRoute = createApmServerRoute({
   },
 });
 
+const transactionChartsColdstartRateRoute = createApmServerRoute({
+  endpoint:
+    'GET /internal/apm/services/{serviceName}/transactions/charts/coldstart_rate',
+  params: t.type({
+    path: t.type({
+      serviceName: t.string,
+    }),
+    query: t.intersection([
+      t.type({ transactionType: t.string }),
+      t.partial({ transactionName: t.string }),
+      t.intersection([environmentRt, kueryRt, rangeRt, comparisonRangeRt]),
+    ]),
+  }),
+  options: { tags: ['access:apm'] },
+  handler: async (resources) => {
```

Comment:
Can you add an explicit return type here? VS Code should give you a refactor action that will add it.